### ISSUE TITLE
Compile error when using block syntax for inner interfaces

### ISSUE THREAD

**Issue by felixvf** [mirah/mirah#297]
Consider the inner interface org.foo.InnerInterfaceClass.InnerInterface

```java
package org.foo;

public class InnerInterfaceClass {
  public static interface InnerInterface {
    public void foo(Object a);
  }
  
  public static void forward(Object param,InnerInterface i) {
    i.foo(param);
  }
}
```

and consider this inner interface being used as such:

```mirah
import org.foo.InnerInterfaceClass

InnerInterfaceClass.forward("foo") do |param|
  puts param
end
```

Then the compilation should succeed, but fails with:

```
Mirah::MirahError: Internal compiler error: java.lang.UnsupportedOperationException: Multiple abstract methods in org.foo.InnerInterfaceClass$InnerInterface: [<MethodType: name=foo [java.lang.Object] : void>, <MethodType: name=InnerInterface [] : org.foo.InnerInterfaceClass$InnerInterface>] Multiple abstract methods in org.foo.InnerInterfaceClass$InnerInterface: [<MethodType: name=foo [java.lang.Object] : void>, <MethodType: name=InnerInterface [] : org.foo.InnerInterfaceClass$InnerInterface>]. Rerun in verbose mode for stacktrace.
```

Apparently, some kind of constructor method seems to be synthesized which counts as "extra" method and makes mirahc fail.

**Comment by baroquebobcat** [comment]
Thanks for the report. That does sound like what's going on.

**Comment by felixvf** [comment]
This is now fixed by #298.

**Comment by baroquebobcat** [comment]
Thanks. Merged.
